examples/error_handling.rs

```diff
@@ -0,0 +1,26 @@
+use tide::utils::After;
+use tide::{Request, Response, Result, StatusCode};
+
+#[async_std::main]
+async fn main() -> Result<()> {
+    tide::log::start();
+    let mut app = tide::new();
+
+    app.middleware(After(|mut res: Response| async {
+        if let Some(err) = res.downcast_error::<tide::http::Error>() {
```

Comment:
I think it would be useful to have a second if branch here that shows downcasting to an external error type like an io error or parse error or something like that to demonstrate it's possible